### PR TITLE
HCX-119:HCX Participant Code and End Point URL validations

### PR DESCRIPTION
**Pull request by tarento-dharmateja**

### CODE REVIEW

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java [hunk 1]
```diff
@@ -113,4 +111,12 @@ private ParticipantResponse errorResponse(ErrorCodes code, String message, Throw
         return resp;
     }
 
+    private HttpEntity<ParticipantResponse> validateParticipant (Map<String, Object> body) {
+        List<String> notAllowedUrls = env.getProperty(HCX_NOT_ALLOWED_URLS, List.class, new ArrayList<String>());
```
Comment (shiva-rakshith) [inline hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java:115]: Remove return null and add method return type to void

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java [hunk 2]
```diff
@@ -60,6 +57,7 @@ public ResponseEntity<Object> participantCreate(@RequestHeader HttpHeaders heade
     @RequestMapping(value = "/search", method = RequestMethod.POST)
     public ResponseEntity<Object> participantSearch(@RequestBody Map<String, Object> requestBody) throws Exception {
         String url =  registryUrl + "/api/v1/Organisation/search";
+        validateParticipant(requestBody);
```
Comment (shiva-rakshith) [inline hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java:60]: validation should be added to participantUpdate, for participant search it is not required.

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java [hunk 3]
```diff
@@ -113,4 +111,12 @@ private ParticipantResponse errorResponse(ErrorCodes code, String message, Throw
         return resp;
     }
 
+    private HttpEntity<ParticipantResponse> validateParticipant (Map<String, Object> body) {
```
Comment (shiva-rakshith) [inline hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java:114]: Update the method to validateEndpointUrl

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java (hunk 2 above)
Comment (tarento-dharmateja) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java:60]: done

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java (hunk 3 above)
Comment (tarento-dharmateja) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java:114]: done

File: hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java (hunk 1 above)
Comment (tarento-dharmateja) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java:115]: while using new ResponseEntity<> inside method unable to change method name void.

Comment (shiva-rakshith) [reply hcx-apis/src/main/java/org/swasth/hcx/controllers/v1/ParticipantController.java:115]: Ok

**Review by shiva-rakshith — APPROVED** [review]

**Pull request merged**